src/app/organizations/organization/organization-starring/organization-starring.component.ts

```diff
@@ -0,0 +1,144 @@
+/*
+ *    Copyright 2019 OICR
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+import { Component, EventEmitter, Input, OnChanges, OnDestroy, OnInit, Output } from '@angular/core';
+import { Organization, User} from '../../../shared/swagger';
+import { TrackLoginService } from '../../../shared/track-login.service';
+import { UserQuery } from '../../../shared/user/user.query';
+import { ContainerService } from '../../../shared/container.service';
+import { StarringService } from '../../../starring/starring.service';
+import { StarentryService } from '../../../shared/starentry.service';
+import { StarOrganizationService } from '../../../shared/star-organization.service';
+import { AlertService } from '../../../shared/alert/state/alert.service';
+import { first, takeUntil } from 'rxjs/operators';
+import { Observable, Subject } from 'rxjs';
+import { OrganizationStarringService } from './organization-starring.service';
+import { calculateRate } from '../../../shared/starring';
+
+@Component({
+  selector: 'app-organization-starring',
+  templateUrl: '../../../starring/starring.component.html',
+  styleUrls: ['../../../starring/starring.component.css']
+})
+export class OrganizationStarringComponent implements OnInit, OnDestroy, OnChanges {
+
+  @Input() organization: Organization;
+  @Output() change: EventEmitter<boolean> = new EventEmitter<boolean>();
+  private user: User;
+  public isLoggedIn: boolean;
+  public rate = false;
+  public total_stars = 0;
+  public disable = false;
+  private ngUnsubscribe: Subject<{}> = new Subject();
```

Comment:
Gary already commented on this below,  but you can just have your class `extend Base`, and then you don't need this either.